tests/integ/scala/test_function_suite.py

```diff
@@ -1287,6 +1288,70 @@ def test_to_timestamp(session):
     )
 
 
+@pytest.mark.localtest
+def test_to_time(session):
+    # basic string expr
+    df = TestData.time_primitives1(session)
+    Utils.check_answer(
+        df.select(*[to_time(column) for column in df.columns]),
+        [
+            Row(time(1, 2, 3)),
+            Row(time(22, 33, 44)),
+        ],
+    )
+
+    # string expr with format
+    df = TestData.time_primitives2(session)
+    Utils.check_answer(
+        df.select(*[to_time(column, "HH12.MI-SS PM") for column in df.columns]),
+        [
+            Row(time(13, 2, 3)),
+            Row(time(22, 33, 44)),
+            Row(time(12, 55, 19)),
+        ],
+    )
+
+    # timestamp
+    df = TestData.time_primitives3(session)
+    Utils.check_answer(
+        df.select(*[to_time(column) for column in df.columns]),
+        [
+            Row(time(12, 13, 14)),
+            Row(time(20, 21, 22)),
+            Row(time(21, 20, 19)),
+        ],
+    )
+
+    # variant type
+    df = TestData.time_primitives4(session)
+    Utils.check_answer(
+        df.select(*[to_time(column) for column in df.columns]),
+        [
+            Row(time(1, 2, 3)),
+            Row(time(21, 20, 19)),
+            Row(
+                None,
+            ),
+        ],
+    )
+
+    # variant data
+
+    # invalid input for string expr with format
```

Comment:
`expected_exception = ValueError if local_testing_mode else SnowparkSQLException`?

Reply:
todo: turn on this test

Reply:
updated